components/script/dom/eventtarget.rs

```diff
@@ -401,12 +408,24 @@ impl EventTarget {
                     entries.push(EventListenerEntry {
                         phase: ListenerPhase::Bubbling,
                         listener: EventListenerType::Inline(listener),
+                        once: false
                     });
                 }
             },
         }
     }
 
+    pub fn remove_listener_if_once(&self, ty: &Atom, listener: Rc<EventListener>) {
+        let mut handlers = self.handlers.borrow_mut();
+
+        let listener = EventListenerType::Additive(listener);
+        for entry in handlers.get_mut(ty) {
+            if let Some(position) = entry.iter().position(|e| e.listener == listener && e.once) {
```

Comment:
You can probably use `drain` here, which will result in slightly more performant code.